src/SHAutomation.Core/AutomationElements/Window.Methods.cs

```diff
@@ -63,7 +63,10 @@ public ISHAutomationElement Find(Func<ConditionFactory, ConditionBase> condition
             }
             else
             {
-                control = parent.FindFirstDescendant(conditionFunc, timeout);
+                if (parent.GetCurrentPropertyValue(PropertyId.Register(AutomationType.UIA3, 30005, "Name")) != null)
```

Comment:
@richardSH2 Might be worth changing this to IsOffscreen as Name can be empty, not sure if this would produce null falsely. Id for this is 30022, "IsOffscreen"

![image](https://user-images.githubusercontent.com/48757510/155115136-4b8ec87e-574f-4ffa-9408-d76c4c2f3cb2.png)

Reply:
Hi @ThomasPoucher have pushed a change following your thoughts on the code. Thanks

Reply:
Thanks @richardSH2